### ISSUE TITLE
rms too high

### ISSUE THREAD

**Issue by HannoSpreeuw** [transientskp/pyse#7]
I constructed a 4k*4k image with 409² 100 Jy sources with a Gaussian profile on a regular grid, inspired by the authors of [this paper](https://www.cse.ust.hk/catalac/papers/source_extraction_escience2013.pdf). They use the same size image, with 167000 sources. 
I inserted those sources in the image plane on top of pure Gaussian - so uncorrelated - noise with a std of 1 Jy.

I ran 
`pyse --detection=50 --analysis=50 --grid=4096 --rmsmap --islands SOURCESINSERTED_100Jy.FITS` to get a single noise level - should be close to 1 Jy - and, hopefully 167281 detections.

Unfortunately, the rms value from the rms.fits is 19.5608 Jy. Way too high and way higher than the ground truth of 1 Jy.

Same output with pyse.py from the tkp repo.

**Comment by gijzelaerr** [comment]
ouch!

**Comment by HannoSpreeuw** [comment]
@gijzelaerr I want to assign myself to this issue, but apparently I do not have that permission.

**Comment by gijzelaerr** [comment]
i invited you to become an admin, check your mail plz

**Comment by HannoSpreeuw** [comment]
No clipping is done at all, since the initial standard deviation -from all pixels - is 19.5608 Jy. Very high, because I inserted sources at a high surface density. Somewhat higher than the highest density I used for pyse validation as part of my thesis work. That explains why I did not encounter this before. 

We have a dynamical clipping limit in pyse, based on the number of independent pixels. 
Dynamical clipping limits are good in principle, I think.

The number of independent pixels is equal to the number of pixels divided by the FWHM beam size.
Now the noise I inserted is uncorrelated, so I should set the beamsize equal to 1 such that the number of independent pixels equals the number of pixels, i.e. all pixels are independent:
```--detection=50 --analysis=50 --grid=4096 --bmaj=0.001414711 --bmin=0.001111111 --bpa=0 --rmsmap --islands SOURCESINSERTED_100Jy.FITS```
So this gets you a FWHM (beam size of pi/4) * 0.001414711*0.001111111/0.001111111² = 1

This is the correct way of running pyse on this artificial image with uncorrelated noise, but making sure that pyse uses a beamsize of 1 does not solve my problem, on the contrary. 
kappa - which is erroneously called my_sigma in pyse - becomes even higher - 5.54 - and no clipping occurs, because 5.54*19.5608 is higher than the highest pixel value - 88.5 Jy.

I could fix this by abandoning dynamical clipping limits and setting kappa to, e.g., 1.5. 
This will result in clipping for virtually any image.
This turns out to work very nicely for this particular artificial 4K*4K image.

Now I also tried this for a 4K*4K image without sources, just the uncorrelated Gaussian noise with a std of 1 Jy.
Setting kappa to 1.5 obviously leads to an underestimate of the background noise, which I can fix very accurately by using a correction equal to var_helper in stats.py. var_helper is currently not used.

So I could implement this in a separate branch, but will require serious validation on thousands of images a pull request can be submitted. Although this particular image is pretty large, so a few hundred with various signal to noise levels might suffice.
